### DISCUSSION TITLE
How to access a deleted public git repository?

### DISCUSSION THREAD

**Discussion by nitinjs** [community/community#135755] [Other Feature Feedback, Questions, & Ideas] [answered]
### Build failing because dependent git repo deleted

Question

### Body

1. How to access a deleted public git repository?
to be more specific I want to download https://github.com/Marqin/dynamicLinker

2. How do I check if the above repo is in code vault? how do I download from code vault?
I have head about "Arctic Code Vault" dont know much about it

anyways thanks

**Comment by erencanbas** [comment, marked as answer]
**Google Cache:** Search for the repository URL on Google using the query site:github.com/Marqin/dynamicLinker. If Google had indexed the repository, there might be a "Cached" link in the search results. Clicking on this will show you the last snapshot Google took of the repository page.

**Wayback Machine:** Visit the Wayback Machine and enter the repository URL. This site archives web pages over time, and if the repository was captured, you could find an older version there.

**Comment by nitinjs** [comment]
In wayback machine cache is there but the code files inside the repo are not cached somehow
https://web.archive.org/web/20201211174520/github.com/Marqin/dynamicLinker

**Comment by erencanbas** [comment]
Unfortunately, there isn't a direct public interface to query whether a specific repository is in the Arctic Code Vault. But you can do the following:

**Check Commit Dates:** If the repository existed and had commits before February 2, 2020, it's likely included in the Arctic Code Vault.

**Contact GitHub Support:** GitHub Support might be able to confirm whether the repository is part of the Arctic Code Vault. You can reach out to them through their [Support page](https://support.github.com/).


Given that the repository URL you provided is no longer available and the Wayback Machine doesn't have the full files cached, these alternatives might be your best bet. Unfortunately, there is no guaranteed method to retrieve code from a deleted repository unless it was archived or copied elsewhere before deletion.

**Comment by nitinjs** [comment]
luckily someone forked the repo
I clicked on forks in wayback cache
and then directly visited the forked path and got the code

thanks for reply anyways
